bindings/sundials4py/cvodes/cvodes.cpp

```diff
@@ -0,0 +1,345 @@
+          {
+            free(fn_table);
+            throw sundials4py::error_returned(
+              "Failed to set user data in CVODE memory");
+          }
+
+          fn_table->fB = nb::cast(fB);
+          return cv_status;
+        });
+
+  m.def("CVodeQuadInitB",
+        [](void* cv_mem, int which,
+           std::function<std::remove_pointer_t<CVQuadRhsFnB>> fQB, N_Vector yQBO)
+        {
+          auto fn_table = get_cvodea_fn_table(cv_mem, which);
+          fn_table->fQB = nb::cast(fQB);
+          return CVodeQuadInitB(cv_mem, which, cvode_fQB_wrapper, yQBO);
+        });
+
+  BIND_CVODEB_CALLBACK(CVodeSetJacFnB, CVLsJacFnB, lsjacfnB,
+                       cvode_lsjacfnB_wrapper, nb::arg("cv_mem"),
+                       nb::arg("which"), nb::arg("jacB").none());
+
+  BIND_CVODEB_CALLBACK2(CVodeSetPreconditionerB, CVLsPrecSetupFnB, lsprecsetupfnB,
+                        cvode_lsprecsetupfnB_wrapper, CVLsPrecSolveFnB,
+                        lsprecsolvefnB, cvode_lsprecsolvefnB_wrapper,
+                        nb::arg("cv_mem"), nb::arg("which"),
+                        nb::arg("psetB").none(), nb::arg("psolveB").none());
+
+  BIND_CVODEB_CALLBACK2(CVodeSetJacTimesB, CVLsJacTimesSetupFnB,
+                        lsjactimessetupfnB, cvode_lsjactimessetupfnB_wrapper,
+                        CVLsJacTimesVecFnB, lsjactimesvecfnB,
+                        cvode_lsjactimesvecfnB_wrapper, nb::arg("cv_mem"),
+                        nb::arg("which"), nb::arg("jsetupB").none(),
+                        nb::arg("jtimesB").none());
+
+  BIND_CVODEB_CALLBACK(CVodeSetLinSysFnB, CVLsLinSysFnB, lslinsysfnB,
+                       cvode_lslinsysfnB_wrapper, nb::arg("cv_mem"),
+                       nb::arg("which"), nb::arg("linsysB").none());
+
+  m.def("CVodeInitBS",
+        [](void* cv_mem, int which, std::function<CVRhsStdFnBS> fBS,
+           sunrealtype tB0, N_Vector yB0)
+        {
+          int cv_status = CVodeInitBS(cv_mem, which, cvode_fBS_wrapper, tB0, yB0);
+
+          auto fn_table = cvodea_user_supplied_fn_table_alloc();
+          auto cvb_mem =
+            static_cast<CVodeMem>(CVodeGetAdjCVodeBmem(cv_mem, which));
+          cvb_mem->python = fn_table;
+
+          cv_status = CVodeSetUserDataB(cv_mem, which, cvb_mem);
+          if (cv_status != CV_SUCCESS)
+          {
+            free(fn_table);
+            throw sundials4py::error_returned(
+              "Failed to set user data in CVODE memory");
+          }
+
+          fn_table->fBS = nb::cast(fBS);
+          return cv_status;
+        });
+
+  m.def("CVodeQuadInitBS",
+        [](void* cv_mem, int which, std::function<CVQuadRhsStdFnBS> fQBS,
+           N_Vector yQBO)
+        {
+          auto fn_table  = get_cvodea_fn_table(cv_mem, which);
+          fn_table->fQBS = nb::cast(fQBS);
+          return CVodeQuadInitBS(cv_mem, which, cvode_fQBS_wrapper, yQBO);
+        });
+
+  BIND_CVODEB_CALLBACK(CVodeSetJacFnBS, CVLsJacStdFnBS, lsjacfnBS,
```

Comment:
Should this be and similarly below
```suggestion
  BIND_CVODEB_CALLBACK(CVodeSetJacFnBS, CVLsJacFnBS, lsjacfnBS,
```
See the question above about when to use `Fn` or `StdFn`

Reply:
Should be StdFn for the same reason as noted in the comment below.

Reply:
I see now these are different because they have to deal with arrays of vectors